### ISSUE TITLE
make_fig_1.py  OverflowError: Python int too large to convert to C long

### ISSUE THREAD

**Issue by mylq** [bits-back/bits-back#8]
Hello, an overflow problem occurs to me when I run the make_fig_1.py. Here is the traceback:

runfile('D:/Code/Project/bits-back-master/make_fig_1.py', wdir='D:/Code/Project/bits-back-master')
```
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
Downloading http://[2001:252:0:2::112]:8081/IXCa9bee7c9d834e0e4d14e3b433c66e97e/exdb/mnist/train-images-idx3-ubyte.gz to data/mnist\MNIST\raw\train-images-idx3-ubyte.gz
  0%|          | 0/9912422 [00:00<?, ?it/s]
Extracting data/mnist\MNIST\raw\train-images-idx3-ubyte.gz to data/mnist\MNIST\raw

Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz to data/mnist\MNIST\raw\train-labels-idx1-ubyte.gz
  0%|          | 0/28881 [00:00<?, ?it/s]
Extracting data/mnist\MNIST\raw\train-labels-idx1-ubyte.gz to data/mnist\MNIST\raw

Downloading http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz
Downloading http://[2001:252:0:2::112]:8081/IXC4850d335c3a12bee99feaea3a5415a6f/exdb/mnist/t10k-images-idx3-ubyte.gz to data/mnist\MNIST\raw\t10k-images-idx3-ubyte.gz
  0%|          | 0/1648877 [00:00<?, ?it/s]
Extracting data/mnist\MNIST\raw\t10k-images-idx3-ubyte.gz to data/mnist\MNIST\raw

Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz to data/mnist\MNIST\raw\t10k-labels-idx1-ubyte.gz
  0%|          | 0/4542 [00:00<?, ?it/s]
Extracting data/mnist\MNIST\raw\t10k-labels-idx1-ubyte.gz to data/mnist\MNIST\raw

C:\ProgramData\Anaconda3\lib\site-packages\torchvision\datasets\mnist.py:67: UserWarning: test_data has been renamed data
  warnings.warn("test_data has been renamed data")
Traceback (most recent call last):

  File "D:\Code\Project\bits-back-master\make_fig_1.py", line 38, in <module>
    bb_ans_compressed = bytes_to_img(rans.flatten(state).tobytes())

  File "D:\Code\Project\bits-back-master\rans.py", line 107, in flatten
    return np.asarray(out, dtype=np.uint32)

  File "C:\ProgramData\Anaconda3\lib\site-packages\numpy\core\_asarray.py", line 102, in asarray
    return array(a, dtype, copy=False, order=order)

OverflowError: Python int too large to convert to C long
```

**Comment by j-towns** [comment]
It seems this issue only affects certain systems (see discussion here: https://github.com/numpy/numpy/issues/6289). I've attempted a fix in https://github.com/j-towns/bits-back/commit/99a5d2d1e25b790dc796e36cdcdbbc276d712cd3, can you see whether that change fixes the issue? If so I'll merge.

**Comment by mylq** [comment]
> It seems this issue only affects certain systems (see discussion here: [numpy/numpy#6289](https://github.com/numpy/numpy/issues/6289)). I've attempted a fix in [j-towns@99a5d2d](https://github.com/j-towns/bits-back/commit/99a5d2d1e25b790dc796e36cdcdbbc276d712cd3), can you see whether that change fixes the issue? If so I'll merge.

Thanks for your reply, It works after I replace "out, msg = [msg[0] >> 32, msg[0]], msg[1]" with "out, msg = [msg[0] >> 32, msg[0] & tail_bits], msg[1]" in the line 106 of rans.py file.
By the way, it also works in j-towns/bits-back branch without any changes.
